### ISSUE TITLE
Support for .WRITE method in UPDATE statement

### ISSUE THREAD

**Issue by FlipSs** [linq2db/linq2db#2140]
Hello,

For example, I have a table with an NVARCHAR(MAX) column, and I want to append bytes to this column.
I can create a custom object, let's call it IList<T>, and map it to a binary serializer to serialize this object into a byte array.
This object may have a method like this:
```
[Sql.Expression("{0} + {1}")]
IList<T> Add(T item);
```

And I want to use this method like so:
```
connection.SomeTable
      .AsUpdatable()
      .Set(x => x.Items, x => x.Items.Add(item))
```

But according to the documentation, there is a more efficient way to append bytes is to use the .WRITE method.
https://docs.microsoft.com/en-us/previous-versions/sql/sql-server-2008-r2/ms177523(v=sql.105)

But I cannot find a way to use this method with IUpdatable, because SqlSetExpression always puts '=' between the column name and the expression.

Is there any other way to use .WRITE method besides building entire query manually?

### Environment details
linq2db version: 2.9.6
Database Server: Sql Server
.NET Framework: .Net Core 3.1, net472

**Comment by MaceWindu** [comment]
Don't really see any hack with current implementation that could help here.

**Comment by sdanyliv** [comment]
I think we have to support this via the following API.
```cs
connection.SomeTable
      .AsUpdatable()
      .Set(x => $”{x.Items}.WRITE({item}, {offset}, {length})”)
      .Update();
```
It should  be no not too much work to do.

**Comment by MaceWindu** [comment]
Implemented in preview2 #2141
